src/models/blockchain/transaction/validity/removal_records_integrity.rs

```diff
@@ -380,13 +403,12 @@ impl ConsensusProgram for RemovalRecordsIntegrity {
         );
 
         // iterate over all input UTXOs
-        let mut all_aocl_indices: Vec<u64> = Vec::new();
         let mut input_index: usize = 0;
         while input_index < input_utxos.len() {
             let utxo: &Utxo = &input_utxos[input_index];
             let utxo_hash = Hash::hash(utxo);
-            let msmp: &MsMembershipProof = &rriw.membership_proofs[input_index];
-            let removal_record: &RemovalRecord = &rriw.removal_records[input_index];
+            let msmp = &rriw.partial_membership_proofs[input_index];
+            let claimed_absolute_indices = &rriw.removal_records[input_index].absolute_indices;
```

Comment:
I included the explicit type indications in order to make the rust code compiler-friendly. Since we are writing everything by hand, it is less important. But there is a reason why we might want to keep the rust code compiler-friendly: if we want to verify, as an extra precaution to boost our confidence in the soundness of the consensus programs, then we can test the handwritten against the compiled tasm code.